app/controllers/api/cloud_networks_controller.rb

```diff
@@ -9,5 +9,22 @@ def delete_resource(type, resource_id, _data = {})
       task_id = cloud_network.delete_cloud_network_queue(User.current_user.userid)
       action_result(true, "Deleting #{cloud_network.name}", :task_id => task_id)
     end
+
+    def options
+      return super unless params.keys.include_any?(%w[id ems_id])
+
+      if params.key?("ems_id")
+        ems = resource_search(params["ems_id"], :ext_management_systems, ExtManagementSystem)
+        raise BadRequestError, "No Cloud Network support for - #{ems.class}" unless defined?(ems.class::CloudNetwork)
+
+        klass = ems.class::CloudNetwork
```

Comment:
`CloudNetwork.class_by_ems(ems)` is the preferred way of getting the class name

Reply:
Which line is this referring to?

Reply:
this one

```suggestion
        klass = CloudNetwork.class_by_ems(ems)
```